src/store/bookmarkStore.js

```diff
@@ -0,0 +1,19 @@
+import {makeAutoObservable} from 'mobx';
+
+class BookmarkStore {
+  bookmarks = [];
+  isBookmarkSet;
+  isButtonPressed = false;
+
+  constructor() {
+    makeAutoObservable(this);
+  }
+
+  setIsBookmarkSet = (bool) => this.isBookmarkSet = bool;
+
+  setIsButtonPressed = (bool) => this.isButtonPressed = bool;
+
+  setBookmarks = (bookmarks) => this.bookmarks = bookmarks;
```

Comment:
Let's try to not use inline functions like this. Why:
```
if (setIsBookmarkSet(true)) {
  console.log(123)
}
```
this will log 123 in the console, because the result of setIsBookmarkSet === true. This isn't your initial intention (to return true or false), the goal of this function is to change isBookmarkSet, so the function must not return anything (undefined). So write it like:
```
setIsBookmarkSet = (bool) => { 
  this.isBookmarkSet = bool;
}
```